src/Umbraco.Core/Security/BackOfficeUserManager.cs

```diff
@@ -238,5 +237,244 @@ public override bool SupportsUserPhoneNumber
         /// Gets/sets the default back office user password checker
         /// </summary>
         public IBackOfficeUserPasswordChecker BackOfficeUserPasswordChecker { get; set; }
+
+        public override Task<IdentityResult> SetLockoutEndDateAsync(int userId, DateTimeOffset lockoutEnd)
+        {
+            var result = base.SetLockoutEndDateAsync(userId, lockoutEnd);
+
+            // The way we unlock is by setting the lockoutEnd date to the current datetime
+            if (result.Result.Succeeded && lockoutEnd >= DateTimeOffset.UtcNow)
+                RaiseAccountLockedEvent(userId);
+            else
+                RaiseAccountUnlockedEvent(userId);
+
+            return result;
+        }
+
+        public override Task<IdentityResult> AccessFailedAsync(int userId)
+        {
+            var result = base.AccessFailedAsync(userId);
+
+            //Slightly confusing: this will return a Success if we successfully update the AccessFailed count
+            if (result.Result.Succeeded)
+                RaiseLoginFailedEvent(userId);
+
+            return result;
+        }
+
+        public override Task<IdentityResult> ChangePasswordAsync(int userId, string currentPassword, string newPassword)
+        {
+            var result = base.ChangePasswordAsync(userId, currentPassword, newPassword);
+            if (result.Result.Succeeded)
+                RaisePasswordChangedEvent(userId);
+            return result;
+        }
+
+        public override async Task<IdentityResult> ResetAccessFailedCountAsync(int userId)
+        {
+            var user = ApplicationContext.Current.Services.UserService.GetUserById(userId);
+
+            if (user == null)
+            {
+                throw new ProviderException(string.Format("No user with the id {0} found", userId));
+            }
+
+            if (user.FailedPasswordAttempts > 0)
+            {
+                user.FailedPasswordAttempts = 0;
+                ApplicationContext.Current.Services.UserService.Save(user);
```

Comment:
There should be no singleon usages here. I'm unsure why we are totally overriding this method and not calling `base.ResetAccessFailedCountAsync(userId);` ? Also note that in ASP.NET identity when working with the UserManager, this talks to the database by the `UserStore`, the user manager doesn't know about the UserService - that is hidden behind our implementation of ASP.NET Identity's `UserStore`

Reply:
I've pushed a fix for this - i realize why we don't call the base class and that's because it doesn't provide us with control over when the value is reset. However, this method should use ASP.NET Identity APIs (basically use the exact same logic as the underlying class) so I've pushed those changes.